napari/components/_layer_slicer.py

```diff
@@ -151,26 +178,36 @@ def slice_layers_async(
             else:
                 sync_layers.append(layer)
 
-        # create task for slicing of each request/layer
-        task = self._executor.submit(self._slice_layers, requests)
+        # create one task for all requests
+        task = None
+        if len(requests) > 0:
+            task = self._executor.submit(self._slice_layers, requests)
 
-        # submit the sync layers (purposefully placed after async submission)
-        for layer in sync_layers:
-            layer._slice_dims(dims.point, dims.ndisplay, dims.order)
+            # store task for cancellation logic
+            # this is purposefully done before adding the done callback to ensure
+            # that the task is added before the done callback can be executed
+            with self._lock_layers_to_task:
+                self._layers_to_task[tuple(requests)] = task
 
-        # store task for cancellation logic
-        # this is purposefully done before adding the done callback to ensure
-        # that the task is added before the done callback can be executed
-        with self._lock_layers_to_task:
-            self._layers_to_task[tuple(requests)] = task
+            task.add_done_callback(self._on_slice_done)
 
-        task.add_done_callback(self._on_slice_done)
+        # slice the sync layers after async submission so that async
+        # tasks can potentially run concurrently
+        for layer in sync_layers:
+            logger.debug('Sync slicing: %s', layer)
+            if _refresh_sync:
```

Comment:
This adds to the ugliness of this method. It is intended to be temporary while we still have some layers that are not async-sliceable and while `Layer.refresh` is still around. A maybe slightly better alternative is to pass `_refresh_sync` through to `_slice_dims` to avoid the early return there.

Reply:
Yes, but I'd vote not to do it in this PR. I think the best time to do it is when once we've replaced all internal usage of it.